docs/prometheus.rst

```diff
@@ -0,0 +1,39 @@
+.. highlight:: python
+
+
+==========
+Prometheus
+==========
+
+Talisker provides optional `prometheus_client` integration and configuration.
+
+Installation
+------------
+
+The package supports extras args to install `prometheus_client`::
+
+    $ pip install talisker[prometheus]
+
+Configuration
+-------------
+
+`prometheus_client` integration needs no extra configuration if running
+in single process mode.
+
+If the app is running in `multiprocess mode <https://github.com/prometheus/client_python#multiprocess-mode-gunicorn>`_
+(ie. with multiple workers), the `prometheus_multiproc_dir` envvar should be set
+to a writable directory that can be used for metrics. If this envvar is not found,
+a temporary directory will be created and used instead.
```

Comment:
Non-random temp files and directories are subject to all kinds of timing attacks... We could probably use some smart deterministic algorithm, but not sure it's worth. Metrics will reset on reboots anyways, and it may not be as serious as we thought (see [this](https://www.robustperception.io/how-does-a-prometheus-counter-work/)).

If the envvar is not set, `prometheus_client` uses a simple `Lock` to update metrics in memory; so it should be thread-safe. If the envvar is set, it uses metrics protected by a mutex backed by a per-process mmaped file ([source code](https://github.com/prometheus/client_python/blob/master/prometheus_client/core.py)).